tools/fv_io.F90

```diff
@@ -260,12 +260,21 @@ subroutine  fv_io_register_restart(Atm)
        call register_restart_field(Atm%Fv_restart_tile, 'v', Atm%v, dim_names_4d2)
 
        if (.not.Atm%flagstruct%hydrostatic) then
-          call register_restart_field(Atm%Fv_restart_tile,  'W', Atm%w, dim_names_4d3)
-          call register_restart_field(Atm%Fv_restart_tile,  'DZ', Atm%delz, dim_names_4d3)
-          if ( Atm%flagstruct%hybrid_z ) then
-             call register_restart_field(Atm%Fv_restart_tile,  'ZE0', Atm%ze0, dim_names_4d3)
+          if (Atm%flagstruct%make_nh) then ! Hydrostatic restarts dont have these variables
+               call register_restart_field(Atm%Fv_restart_tile,  'W', Atm%w, dim_names_4d3, is_optional=.true.)
+               call register_restart_field(Atm%Fv_restart_tile,  'DZ', Atm%delz, dim_names_4d3, is_optional=.true.)
+               if ( Atm%flagstruct%hybrid_z ) then
```

Comment:
One of the things we will need to clean out soon is the hybrid_z option. I'm getting tired of answering questions about unsupported/unmaintained/undocumented features.

Reply:
Perhaps this can be handled as part of the next public release, if not before.